content/dev/dev-guide.md

```diff
@@ -0,0 +1,325 @@
+
+ - First we need to create a directory for the urbit binary, we'll create this in your home directory.
+   - ```bash
+     mkdir ~/urbit-dev
+     ```
+ - Next we're going to pull down the latest urbit binary from urbit.org from the newly created directory.
+    - ```bash
+      cd ~/urbit-dev
+      curl -JLO https://urbit.org/install/mac/latest
+      ```
+  - Now we'll untar the contents of the download into our newly created directory.
+    - ```bash 
+      tar -zxvf ./darwin.tgz --strip=1
+      ```
+  - With the urbit binary installed we're ready for the basic dev workflow.
+
+## Development Basics
+
+### Fake Ships
+
+#### What is a "fake" ship?
+
+ - For local development we'll create a 'fake' ship, this is a local ship that is not connected to the urbit network. This means we don't need a real ID (eg comet, planet) and can delete and recreate the ship whenever we want to reset things.
+
+#### What is a "pill"?
+
+ - A pill file contains the boot sequence, if it's not specified the livenet pill is used by default. 
+ - Since OS updates propogate through the network, it's possible the default pill file could be slightly out of date.
+ - We can manually get the most up to date pill file for local development by downloading it and using it when starting our fake ship.
+   - For fake development ships not talking to the regular network this works well.
+   - We don't do this for 'real' ships on the actual network because if you're using a pill that's newer than what your sponsor has (the ship that you get updates from) your sponsor will think you're on an older version.
+   - When this happens your sponsor will attempt to upgrade you, but will actually be downgrading you to their verison.
+   - There are compiled and uncompiled pills. They are differentiated by a naming convention that adds `-brass` to the compiled pill name. For example, `multi.pill` is the name of a compiled pill and the name of its uncompiled version is `multi-brass.pill`. For the purposes of this guide we'll use compiled pills because they boot faster. An uncompiled pill lets you audit the source code if you need to.
+ - In summary, if you're using a fake ship -> get the latest pill, if you're starting a real ship don't worry about pills.
+
+#### Creating a fake ship
+
+ - We can create a local fake ship with the below command. This will create a fake ship named zod in our `urbit-dev` directory using the newest `multi.pill` from the urbit code we cloned earlier. This will take a couple of minutes and generate some output in the terminal.
+   - ```bash
+     ./urbit -F zod -B ~/urbit-git/bin/multi.pill
+     ```
+ - When it's finished you'll see a command prompt like `~zod:dojo>` - this is the urbit dojo where you can enter and run urbit commands.
+ - To exit the dojo type `|exit` or `ctrl-D`
+
+#### Fake ship networking
+
+ - Fake ships on the same machine can automatically talk to each other. Let's create a second fake ship named `~bus` and confirm this.
```

Comment:
There are rules about routing tho, correct?  I.e. you need to make sure two planets are from same star.

Reply:
hmm yeah apparently, I didn't realise that

Reply:
added a note